x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/FrozenIndexInput.java

```diff
@@ -83,13 +57,13 @@ public FrozenIndexInput(
             0L,
             0L,
             fileInfo.length(),
+            new CacheFileReference(directory, fileInfo.physicalName(), fileInfo.length()),
```

Comment:
I'm a bit torn on the file's length that is passed here. It means that we'll create a potentially large file in the cache service (and cause potential evictions) while we know it will only be used to cache at most `directory.getBlobCacheByteRange(name, fileInfo.length()).length()` bytes. We could maybe affine this?

Reply:
Note that CacheService is always supposed to be unbounded (and that's how it should be, the goal is to remove the undocumented setting to even make this configurable, and remove the complexity around that)

Reply:
Right, I forgot about that point. Also my suggestion was pretty bad since the file length here passed to the sparse file tracker and that would just break things. Sorry for the noise.